onnxscript/ir/_linked_list.py

```diff
@@ -69,14 +59,13 @@ def erase(self) -> None:
         prev, next_ = self.prev, self.next
         prev.next, next_.prev = next_, prev
         # Detach the value
-        self.value._link_box = None  # pylint: disable=protected-access
         self.value = None
 
     def __repr__(self) -> str:
         return f"_LinkBox({self.value!r}, erased={self.erased}, prev={self.prev.value!r}, next={self.next.value!r})"
 
 
-class DoublyLinkedList(Generic[TLinkable], Sequence[TLinkable]):
+class DoublyLinkedHashList(Generic[T], Sequence[T]):
```

Comment:
May be DoublyLinkedSet is a better name (focused on the intended semantics that values occur only once in the ordered list ... adding "Hash" may not help, given the ambiguity in its role ("DoublyLinked" is the primary implementation mechanism, Hash gives random-access by value).

Reply:
The only concern I have for a "set" is that we would then assume set operations like `add` and `union` etc. The apis we provide resemble that of a list more. Of course that is not a huge concern, since it is a data structure we consume internally.

Reply:
Renamed to DoublyLinkedSet

Reply:
> The only concern I have for a "set" is that we would then assume set operations like `add` and `union` etc. The apis we provide resemble that of a list more. Of course that is not a huge concern, since it is a data structure we consume internally.

Right. I just feel it helps to clarify the uniqueness of elements. Not a big deal either way once users understand what it is supposed to be. (HashList, on the other hand, might convey a linked-list implementation used for overflow-chains in a hash-table. I guess everything has some minor drawback.)